code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -147,31 +147,31 @@ Pipelines + Other Objects -> Pipe network
 				"<span class='italics'>You hear ratchet.</span>")
 			investigate_log("was <span class='warning'>REMOVED</span> by [key_name(usr)]", "atmos")
 
-			//You unwrenched a pipe full of pressure? let's splat you into the wall silly.
+			//You unwrenched a pipe full of pressure? Let's splat you into the wall, silly.
 			if(unsafe_wrenching)
-				unsafe_pressure_release(user,internal_pressure)
+				unsafe_pressure_release(user, internal_pressure)
 			Deconstruct()
 
 	else
 		return ..()
 
 
-//Called when an atmospherics object is unwrenched while having a large pressure difference
-//with it's locs air contents.
-/obj/machinery/atmospherics/proc/unsafe_pressure_release(mob/user,pressures)
+// Throws the user when they unwrench a pipe with a major difference between the internal and environmental pressure.
+/obj/machinery/atmospherics/proc/unsafe_pressure_release(mob/user, pressures = null)
 	if(!user)
 		return
-
 	if(!pressures)
 		var/datum/gas_mixture/int_air = return_air()
 		var/datum/gas_mixture/env_air = loc.return_air()
-		pressures = int_air.return_pressure()-env_air.return_pressure()
+		pressures = int_air.return_pressure() - env_air.return_pressure()
+
+	var/dir = get_dir(src, user)
+	if(!dir)
+		dir = rand(1,4)

```

Comment:
This won't work, the cardinals are 1, 2, 4, and 8. Change this to `pick(cardinals) `.